### PR TITLE
Removes wrapper class for Base64

### LINKED ISSUE #2032: Remove Base64Wrapper class
#### Software and hardware versions 
Collect v1.13

#### Problem description
[Base64Wrapper](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/utilities/Base64Wrapper.java) was added for accessing [Base64](https://developer.android.com/reference/android/util/Base64.html) functionality in API 7. But now the minSdkVersion is 16 so this wrapper class can be removed. Also, since in the current wrapper class reflection is being used which is slower than normal function call.

@opendatakit-bot claim

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Closes #2032

#### What has been done to verify that this works as intended?
Tried saving, editing, and closing/opening an unsaved form on a physical device.

#### Why is this the best possible solution? Were any other approaches considered?
Since the minSdkVersion is 16 so the wrapper class can be removed as Base64 was added in API 8.
Also, the current implementation uses reflection.

#### Are there any risks to merging this code? If so, what are they?
None

#### Do we need any specific form for testing your changes? If so, please attach one.
`All Widgets Form`

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java [hunk 1]
```diff
@@ -97,6 +98,7 @@
             "base64EncryptedElementSignature";
     private static final String NEW_LINE = "\n";
     private static final String ENCRYPTION_PROVIDER = "BC";
+    public static final int FLAGS = 2; // NO_WRAP
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java]: screen shot 2018-03-18 at 6 56 28 pm

**Review by dcbriccetti — CHANGES_REQUESTED** [review]
QA can start. The change I asked for won’t break anything.

File: collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java [hunk 2]
```diff
@@ -154,8 +154,7 @@ private EncryptionUtils() {
                 byte[] pkEncryptedKey = pkCipher.doFinal(key);
                 String alg = pkCipher.getAlgorithm();
                 Timber.i("AlgorithmUsed: %s", alg);
-                base64RsaEncryptedSymmetricKey = wrapper
-                        .encodeToString(pkEncryptedKey);
+                base64RsaEncryptedSymmetricKey = Base64.encodeToString(pkEncryptedKey, FLAGS);
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java]: While we’re here let’s change `AlgorithmUsed` to `Algorithm used`.

File: collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java]: Thanks @dcbriccetti 
Done

File: collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java (hunk 2 above)
Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/utilities/EncryptionUtils.java]: Done

**Review by dcbriccetti — APPROVED** [review]

**Comment by dcbriccetti** [comment]
Bot, why are you fighting me‽ Please confer with your maker (@yanokwa ?).

**Comment by shobhitagarwal1612** [comment]
This bot has gone rogue :laughing:

**Comment by dcbriccetti** [comment]
>This bot has gone rogue

C-3PO it’s not. :grin:

**Comment by lognaturel** [comment]
The QA step will be to collect an encrypted form and verify it can be decrypted as expected.

@shobhitagarwal1612 one other small thing -- can you please revert the indentation change to some of the string resource files? Thanks!

**Comment by shobhitagarwal1612** [comment]
>can you please revert the indentation change to some of the string resource files

@lognaturel Done.

Actually, I used Translations Editor to remove the strings at once, so forgot to double check before pushing

**Comment by lognaturel** [comment]
> I used Translations Editor to remove the strings at once

Thanks for letting me know! I had seen this kind of indentation mismatch before and wasn't sure why it happened.

**Review by lognaturel — APPROVED** [review]

**Comment by mmarciniak90** [comment]
**Tested with success!**

Verified on Android: 4.1, 4.2, 4.4, 5.1, 6.0, 7.0, 8.0.

Verified cases:
* filled encrypted form `encrypted-form.xml`
* edit for was not possible
* upload submissions to Aggregate
* export and decrypt submission with Briefcase
* additionally tested `All widget` form

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Pull request merged**